restapi/src/main/java/io/stargate/web/resources/v2/schemas/UserDefinedTypesResource.java

```diff
@@ -0,0 +1,388 @@
+      @ApiParam(value = "Unwrap results", defaultValue = "false") @QueryParam("raw")
+          final boolean raw,
+      @Context HttpServletRequest request) {
+    return RequestHandler.handle(
+        () -> {
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, getAllHeaders(request));
+          Keyspace keyspace = authenticatedDB.getDataStore().schema().keyspace(keyspaceName);
+          if (keyspace == null) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(
+                    new Error(
+                        "keyspace does not exists", Response.Status.BAD_REQUEST.getStatusCode()))
+                .build();
+          }
+
+          db.getAuthorizationService()
+              .authorizeSchemaRead(
+                  authenticatedDB.getAuthenticationSubject(),
+                  Collections.singletonList(keyspaceName),
+                  null,
+                  SourceAPI.REST,
+                  ResourceKind.TYPE);
+
+          UserDefinedTypeResponse udtResponse =
+              mapUdtAsResponse(authenticatedDB.getType(keyspaceName, typeName));
+          return Response.ok(
+                  Converters.writeResponse(raw ? udtResponse : new ResponseWrapper<>(udtResponse)))
+              .build();
+        });
+  }
+
+  @Timed
+  @POST
+  @ApiOperation(
+      value = "Create a user defined type (UDT)",
+      notes = "Add a user defined type (udt) in a specific keyspace.",
+      response = Map.class,
+      code = 201)
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 201, message = "Created", response = Map.class),
+        @ApiResponse(
+            code = 400,
+            message = "Bad Request, the input is not well formated",
+            response = Error.class),
+        @ApiResponse(
+            code = 401,
+            message = "Unauthorized, token is not valid or not enough permissions",
+            response = Error.class),
+        @ApiResponse(
+            code = 409,
+            message = "Conflict, the object may already exist",
+            response = Error.class),
+        @ApiResponse(code = 500, message = "Internal server error", response = Error.class)
+      })
+  public Response createType(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam(HEADER_TOKEN_AUTHENTICATION)
+          final String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(value = "", required = true) @NotNull final UserDefinedTypeAdd udtAdd,
+      @Context final HttpServletRequest request) {
+
+    return RequestHandler.handle(
+        () -> {
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, getAllHeaders(request));
```

Comment:
For all of these can you please call `getRestDataStoreForToken` instead?
https://github.com/stargate/stargate/pull/911/files#diff-ec4e46205465471675a601af1264a81e773d5a8cc487e2ddae8abaf687548e09R103

Reply:
Done.